lib/Doctrine/DBAL/Portability/Statement.php

```diff
@@ -176,6 +176,7 @@ protected function fixRow($row, bool $iterateRow, bool $fixCase)
         }
 
         if ($fixCase) {
+            assert($this->case !== null);
```

Comment:
Here we could use the same approach as with private methods above, however, this one is protected. We should see what prevents this class from being finalized (https://github.com/doctrine/dbal/pull/3590).

Reply:
What should we do here then? There are only 4 `assert()` left in this PR; do you want to find a solution to get rid of each of them individually, or is it OK if they stay?

Reply:
Let's leave those assertions for now. Please rebase and squash.

Reply:
Rebased & squashed!